opentelemetry-sdk/src/logs/log_emitter.rs

```diff
@@ -239,12 +239,138 @@ impl opentelemetry::logs::Logger for Logger {
+                    "test_service",
+                )])),
+            })
+            .build();
+        assert_resource(&custom_config_provider, SERVICE_NAME, Some("test_service"));
+        assert_eq!(custom_config_provider.config().resource.len(), 1);
+
+        // If `OTEL_RESOURCE_ATTRIBUTES` is set, read them automatically
+        temp_env::with_var(
+            "OTEL_RESOURCE_ATTRIBUTES",
+            Some("key1=value1, k2, k3=value2"),
+            || {
+                let env_resource_provider = super::LoggerProvider::builder().build();
+                assert_resource(
+                    &env_resource_provider,
+                    SERVICE_NAME,
+                    Some("unknown_service"),
+                );
+                assert_resource(&env_resource_provider, "key1", Some("value1"));
+                assert_resource(&env_resource_provider, "k3", Some("value2"));
+                assert_telemetry_resource(&env_resource_provider);
+                assert_eq!(env_resource_provider.config().resource.len(), 6);
+            },
+        );
+
+        // When `OTEL_RESOURCE_ATTRIBUTES` is set and also user provided config
+        temp_env::with_var(
+            "OTEL_RESOURCE_ATTRIBUTES",
+            Some("my-custom-key=env-val,k2=value2"),
+            || {
+                let user_provided_resource_config_provider = super::LoggerProvider::builder()
+                    .with_config(Config {
+                        resource: Cow::Owned(Resource::default().merge(&mut Resource::new(vec![
+                            KeyValue::new("my-custom-key", "my-custom-value"),
+                        ]))),
+                    })
+                    .build();
+                assert_resource(
+                    &user_provided_resource_config_provider,
+                    SERVICE_NAME,
+                    Some("unknown_service"),
+                );
+                assert_resource(
+                    &user_provided_resource_config_provider,
+                    "my-custom-key",
+                    Some("my-custom-value"),
+                );
+                assert_resource(
+                    &user_provided_resource_config_provider,
+                    "k2",
+                    Some("value2"),
+                );
+                assert_telemetry_resource(&user_provided_resource_config_provider);
+                assert_eq!(
+                    user_provided_resource_config_provider
+                        .config()
+                        .resource
+                        .len(),
+                    6
+                );
+            },
+        );
+
+        // If user provided a resource, it takes priority during collision.
+        let no_service_name = super::LoggerProvider::builder()
+            .with_config(Config {
+                resource: Cow::Owned(Resource::empty()),
+            })
+            .build();
+
+        assert_resource(&no_service_name, SERVICE_NAME, None);
```

Comment:
nit - this test is redundant, and can be removed. As we are checking the length in the next test.

Reply:
make sense to me https://github.com/open-telemetry/opentelemetry-rust/pull/1646/commits/7947daa611d98cdc6af89c10996936df865a1a39